### PR TITLE
refactored inheritable transfomrations api

### PR DESCRIPTION
**Pull request by saiskee**
Refactoring the `inheritableTransformations` API that I introduced in [this PR](https://github.com/solo-io/gloo/pull/4163).

Removing `inheritableTransformations` from Route and VHost proto and adding `inheritTransformation` to StagedTransformation options.

### CODE REVIEW

File: changelog/v1.8.0-beta9/inherit-transformations.yaml [hunk 1]
```diff
@@ -2,7 +2,8 @@ changelog:
   - type: NEW_FEATURE
```
Comment (saiskee) [inline changelog/v1.8.0-beta9/inherit-transformations.yaml:2]: Is it okay to repurpose the previous PR's changelog? Not sure I want to create two New feature entries if this is merged in before the next beta release.

Comment (kdorosh) [reply changelog/v1.8.0-beta9/inherit-transformations.yaml:2]: yes, I think that's fine. we can use `/skip-changelog` here

**Review by kdorosh — APPROVED** [review]

**Review by kdorosh — CHANGES_REQUESTED** [review]
need to fix changelog dir

**Review by kdorosh — APPROVED** [review]

**Pull request merged**